src/DataTable.php

```diff
@@ -118,12 +118,15 @@ protected function initColumns()
     {
         $this->_extraColumns = $this->extraColumns;
         if (isset($this->_options['columns'])) {
+          $demoObject = (count($this->_options['data'])) ? $this->_options['data'][0] : null;
```

Comment:
@Radon8472 Does it mean that column names will depend on existence of data?

Reply:
what about to add new widget parameter like `modelClass` and use `\yii\base\StaticInstanceTrait::instance` to create model instance to get translations?

Reply:
I would say either passing an optional new option for modelClass, or yust passing `dataProvider` instead of 'data' (or better check which one exists).

in both cases we should have the chance to get the modelAttributes.

and NO, names do not DEPEND on the existince of data.

If there is data, it takes the model translations, if not, it will yust `camel2words()` the col-name (like it was before)

Reply:
@Radon8472 do you want to try to add support of `dataprovider`?

Reply:
If you like this, I will add this to this PW.

My Idea is quite simple:

- You still can add `data` in the config (to keep the classic behavior)
- I add a new option for `dataProvider`, if it is given, it will be stored, and the missing argument `data` will be filled with `$dataProvider->getModels()`
- and if provider is set, we can load the `modelClass` from it

Reply:
sounds good for me

Reply:
@ZAYEC77 I added and testes the dataProvider implementation.
It works great in both szenarios with dataProvider or with models from `data` property

I added a few minimal errorchecks to affected methods.